### PR TITLE
SAK-48494 SAMIGO: When creating an assessment we should add events on the calendar only for the groups which instructor belongs to

### PR DESCRIPTION
**Pull request by susowrc**
SAMIGO: When creating an assessment we should add events on the calendar only for the groups which instructor belongs to

https://sakaiproject.atlassian.net/browse/SAK-48494

### CODE REVIEW

File: calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java [hunk 1]
```diff
@@ -2693,15 +2693,25 @@ public CalendarEvent addEvent(TimeRange range, String displayName, String descri
...
-					edit.setGroupAccess(groups,true);
-				}
-				catch (PermissionException e)
-				{
-					cancelEvent(edit);
-					throw new PermissionException(m_sessionManager.getCurrentSessionUserId(), eventId(SECURE_ADD), getReference());
+				if(!allowedSelected.isEmpty()) {
+					edit.setGroupAccess(allowedSelected, true);
+					if(!selectedNotAllowed.isEmpty()) {
+						log.warn("Attempting to post to groups you don't belong to "+selectedNotAllowed.toString()+" user= "+m_sessionManager.getCurrentSessionUserId()+" lock="+eventId(SECURE_ADD)+" resource="+getReference());
```
Comment (ern) [inline calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java:2703]: use sfl4j log style

```suggestion
						log.warn("Attempting to post to groups you don't belong to {} user={}  lock={} resource={}", selectedNotAllowed, m_sessionManager.getCurrentSessionUserId(), eventId(SECURE_ADD), getReference());
```

File: calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java [hunk 2]
```diff
@@ -2713,6 +2723,28 @@ public CalendarEvent addEvent(TimeRange range, String displayName, String descri
 			return edit;
 
 		} // addEvent
+
+		/**
+		 * @return only the groups allowed and selected
+		 */
+		private Collection groupAllowedSelect (Collection group, Collection groupAllowed) {
```
Comment (ern) [inline calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java:2730]: need to use types on the Collections (add generics)

File: calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java [hunk 3]
```diff
@@ -2713,6 +2723,28 @@ public CalendarEvent addEvent(TimeRange range, String displayName, String descri
...
+			for (Object it : groupAllowed) {
+				if (group.contains(it)) {selected.add(it);}
+			}
+			return selected;
+		}
+
+		/**
+		 * @return groups not allowed and selected
+		 */
+		private Collection groupNotAllowedSelect (Collection group, Collection groupAllowed) {
```
Comment (ern) [inline calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java:2741]: need to use types on the Collections (add generics)

File: calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java (hunk 2 above)
Comment (ern) [reply calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java:2730]: this method could also be reduced to a lambda

File: calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java [hunk 4]
```diff
@@ -2713,6 +2723,20 @@ public CalendarEvent addEvent(TimeRange range, String displayName, String descri
 			return edit;
 
 		} // addEvent
+
+		/**
+		 * @return only the groups allowed and selected
+		 */
+		private Collection<Group> groupAllowedSelect (Collection<Group> group, Collection<Group> groupAllowed) {
```
Comment (mpellicer) [inline calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java:2730]: Very minor and optional, could we look for a better name?

filterAllowedAndSelectedGroups()

File: calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java [hunk 5]
```diff
@@ -2713,6 +2723,20 @@ public CalendarEvent addEvent(TimeRange range, String displayName, String descri
...
+			return groupAllowed.stream().filter(it -> (group.contains(it))).collect(Collectors.toList());
+		}
+
+		/**
+		 * @return groups not allowed and selected
+		 */
+		private Collection<Group> groupNotAllowedSelect (Collection<Group> group, Collection<Group> groupAllowed) {
```
Comment (mpellicer) [inline calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java:2737]: Very minor and optional, I kinda don't like the function name, better if we provide something more clear like:

filterNotAllowedAndSelectedGroups

**Review by mpellicer — APPROVED** [review]

**Comment by mpellicer** [comment]
Great job @susowrc

**Comment by susowrc** [comment]
Thank you so much!

**Pull request merged**